src/app/components/Forms/CreateProductForm/index.tsx

```diff
@@ -0,0 +1,184 @@
+import { useUser } from '@clerk/nextjs';
+import { yupResolver } from '@hookform/resolvers/yup';
+
+import ButtonSubmit from '../../ButtonSubmit';
+import DefaultInput from '../../DefaultInput';
+import DefaultSelect from '../../DefaultSelect';
+import UploadImageInput from '../../UploadImageInput';
+import styles from './styles.module.css';
+
+export interface createProductForm {
+  setShowDialog: (value: boolean) => void;
+}
+
+export default function CreateProductForm({
+  setShowDialog,
+}: createProductForm) {
+  const [requestError, setRequestError] = useState(false);
+  const [isSubmiting, setIsSubmiting] = useState(false);
+  const [registredWithSucess, setRegistredWithSucess] = useState(false);
+  const [imagePlaceholder, setImagePlaceholder] = useState<string>(``);
+  const [productImage, setProductImage] = useState<ImageProps>(
+    [] as unknown as ImageProps
+  );
+  const { user } = useUser();
+
+  const {
+    handleSubmit,
+    register,
+    control,
+    formState: { errors },
+  } = useForm<ProductData>({
+    resolver: yupResolver(productSchema),
+    mode: `onChange`,
+  });
+
+  const productName = useWatch({
+    control,
+    name: `name`,
+  });
+
+  const onSubmit = async (values: ProductData) => {
+    setIsSubmiting(true);
+
+    const createProductPayload = {
+      ...values,
+      price: Number(values.price),
+      companyId: `${user?.id}`,
+      images: [productImage],
+    };
+
+    try {
+      const createdProduct = await createProduct({
+        ...createProductPayload,
+      });
+
+      if (!createdProduct?.id) {
+        setIsSubmiting(false);
+        return setRequestError(true);
+      }
+    } catch (error) {
+      setRequestError(true);
+    } finally {
+      setShowDialog(false);
+      setIsSubmiting(false);
+    }
+
+    setRegistredWithSucess(true);
+    setIsSubmiting(false);
+  };
+
+  const toBase64 = (file: File | undefined) =>
+    new Promise((resolve, reject) => {
+      // vai sair daqui
+      if (file) {
+        const reader = new FileReader();
+        reader.readAsDataURL(file);
+        reader.onload = () => resolve(reader.result);
+        reader.onerror = reject;
+      }
+    });
+
+  const handleImage = async (event: ChangeEvent<HTMLInputElement>) => {
+    try {
+      const file = event.target.files?.[0];
+
+      const fileParsetToBase64 = await toBase64(file);
+
+      setImagePlaceholder(`${productName}-${Date.now()} ✔️`);
+
+      setProductImage({
+        file: `${fileParsetToBase64}`,
+        name: `${productName}-${Date.now()}`,
+        alt: `Imagem do produto ${productName}`,
+      });
+    } catch (error) {
+      setRequestError(true);
+    }
+  };
+
+  return (
+    <div className={styles.createProductFormContainer}>
+      <form
+        className={styles.createProductForm}
+        onSubmit={handleSubmit(onSubmit)}
+      >
+        <div className={styles.createProductFormcontainerInputs}>
+          <DefaultInput
+            Icon={<FaCarrot />}
```

Comment:
```suggestion
            Icon={<IoFastFoodOutline />}
```
The carrot was just a place holder for lack of a good alternative

Reply:
Hey, this icon is premium, I can't use it, but a find another good alternative

Reply:
MdOutlineFastfood